Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
+                             size_t bytesPerRow, CGDataProviderRef provider,
+                             const CGFloat *decode, bool shouldInterpolate) {
+  RETURN_NULL_IF(((bitsPerComponent == 8) && (bitsPerPixel == 32)));
 
-/**
- @Status Caveat
- @Notes Only 32bpp RGBA source format supported. Returns an 8bpp grayscale alpha mask one-time
-        copy of source bitmap.
-*/
-CGImageRef CGImageMaskCreate(size_t width,
-                             size_t height,
-                             size_t bitsPerComponent,
-                             size_t bitsPerPixel,
-                             size_t bytesPerRow,
-                             CGDataProviderRef provider,
-                             const CGFloat* decode,
-                             bool shouldInterpolate) {
-    FAIL_FAST_HR_IF_FALSE(E_UNEXPECTED, ((bitsPerComponent == 8) && (bitsPerPixel == 32)));
-    FAIL_FAST_HR_IF_FALSE_MSG(E_INVALIDARG,
-                              ((provider == nullptr) || [(NSObject*)provider isKindOfClass:[NSData class]]),
-                              "CGDataProviderRef does not derive from NSData!");
-
-    NSData* dataProvider = (__bridge NSData*)provider;
-    char* pData = (char*)[dataProvider bytes];
-    size_t dataLen = (size_t)[dataProvider length];
-
-    //  Create an 8-bit mask from the data
-    __CGSurfaceInfo surfaceInfo = _CGSurfaceInfoInit(width, height, _ColorGrayscale);
-
-    CGImageRef newImage = new CGBitmapImage(surfaceInfo);
-    char* pNewImage = (char*)newImage->Backing()->LockImageData();
-
-    int pixOut = 0;
-    int outImageBytesPerRow = newImage->Backing()->BytesPerRow();
-
-    for (unsigned y = 0; y < height; y++) {
-        char* rowIn = &pData[y * bytesPerRow];
-        char* rowOut = &pNewImage[pixOut];
-
-        for (unsigned x = 0; x < width; x++) {
-            BYTE r = *rowIn++;
-            BYTE g = *rowIn++;
-            BYTE b = *rowIn++;
-            BYTE a = *rowIn++;
-
-            BYTE alphaOut = (r + g + b) * a / 255 / 3;
-            *rowOut++ = 255 - alphaOut;
-        }
-
-        pixOut += outImageBytesPerRow;
-    }
+  RETURN_NULL_IF(((provider == nullptr) ||
+                  ![(NSObject *)provider isKindOfClass:[NSData class]]));
 
-    newImage->Backing()->ReleaseImageData();
+  NSData *dataProvider = (__bridge NSData *)provider;
 
-    return (CGImageRef)newImage;
-}
+  unsigned char *data = (unsigned char *)[dataProvider bytes];
 
-/**
- @Status Interoperable
-*/
-CGImageAlphaInfo CGImageGetAlphaInfo(CGImageRef img) {
-    int32_t ret;
+  ComPtr<IWICBitmap> image;
+  ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
 
-    if (img) {
-        ret = img->Backing()->BitmapInfo() & kCGBitmapAlphaInfoMask;
-    } else {
-        TraceWarning(TAG, L"CGImageGetAlphaInfo: Null CGImageRef!");
-        ret = 0;
-    }
+  HRESULT status = imageFactory->CreateBitmapFromMemory(
+      width, height, GUID_WICPixelFormatBlackWhite, bytesPerRow,
```

Comment:
>GUID_WICPixelFormatBlackWhite [](start = 21, length = 29)

We have to match the bitsPerPixel with their WIC counterparts:

```
1bpp -> BlackWhite
2bpp -> 2bppGray
4bpp -> 4bppGray
8..
16..
32..
```

Presumably we want fixed point for 32, but it's weird there's no UINT version of it.